.github/workflows/sca.yml

```diff
@@ -86,22 +86,24 @@ jobs:
       - name: Find changed files (for pull request)
         if: ${{ github.event_name == 'pull_request' }}
         run: |
-          git diff origin/$GITHUB_BASE_REF --name-only --diff-filter=ACMRTUXB | grep -E "\.php$" || true
-          echo 'CHANGED_PHP_FILES<<EOF' >> $GITHUB_ENV
-          git diff origin/$GITHUB_BASE_REF --name-only --diff-filter=ACMRTUXB | grep -E "\.php$" || true >> $GITHUB_ENV
-          echo 'EOF' >> $GITHUB_ENV
+          if git diff origin/$GITHUB_BASE_REF --name-only --diff-filter=ACMRTUXB | grep -E "\.php$"; then
+            echo 'CHANGED_PHP_FILES<<EOF' >> $GITHUB_ENV
+            git diff origin/$GITHUB_BASE_REF --name-only --diff-filter=ACMRTUXB | grep -E "\.php$" >> $GITHUB_ENV
+            echo 'EOF' >> $GITHUB_ENV
+          fi
 
       - name: Find changed files (for push)
         if: ${{ github.event_name != 'pull_request' }}
         run: |
-          git diff --name-only --diff-filter=ACMRTUXB HEAD~..HEAD | grep -E "\.php$" || true
-          echo 'CHANGED_PHP_FILES<<EOF' >> $GITHUB_ENV
-          git diff --name-only --diff-filter=ACMRTUXB HEAD~..HEAD | grep -E "\.php$" || true >> $GITHUB_ENV
-          echo 'EOF' >> $GITHUB_ENV
+          if git diff --name-only --diff-filter=ACMRTUXB HEAD~..HEAD | grep -E "\.php$"; then
+            echo 'CHANGED_PHP_FILES<<EOF' >> $GITHUB_ENV
+            git diff --name-only --diff-filter=ACMRTUXB HEAD~..HEAD | grep -E "\.php$" >> $GITHUB_ENV
+            echo 'EOF' >> $GITHUB_ENV
+          fi
 
       - name: Check - phpmd
-        if: ${{ github.env.CHANGED_PHP_FILES }}
-        run: ./dev-tools/vendor/bin/phpmd `echo "$CHANGED_PHP_FILES" | xargs | sed 's/ /,/g'` text phpmd.xml
+        if: ${{ env.CHANGED_PHP_FILES }}
+        run: ./dev-tools/vendor/bin/phpmd `echo "$CHANGED_PHP_FILES" | xargs | sed 's/ /,/g'` ansi phpmd.xml
```

Comment:
should we try to put violation and then use `github` reporter to see how it works?

Reply:
Done.

Reply:
How about using an `if` here to use the `github` format for PRs then `ansi` for push?

Reply:
nice idea!